From 443f7040b07671b36cfe1b70adb0717307cfb8df Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Sat, 4 Sep 2021 19:58:13 +0200
Subject: [PATCH] work around off-by-ones on i386 in uicheck uitests...

<@x1sc0> _rene__, I guess we could use assertAlmostEqual with delta=1 there,
         but maybe once we add it, the next assert also fails for you?

Change-Id: Ic64641ccdf9ca721997b4a842b646cf4b9e30f16
---
 uitest/impress_tests/drawinglayer.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/uitest/impress_tests/drawinglayer.py b/uitest/impress_tests/drawinglayer.py
index 9f193d3cfb41..b1e14861351e 100644
--- a/uitest/impress_tests/drawinglayer.py
+++ b/uitest/impress_tests/drawinglayer.py
@@ -69,7 +69,7 @@ class ImpressDrawinglayerTest(UITestCase):
             self.assertEqual(25199, document.DrawPages[0].getByIndex(0).Size.Width)
             self.assertEqual(2629, document.DrawPages[0].getByIndex(0).Size.Height)
             self.assertEqual(25199, document.DrawPages[0].getByIndex(1).Size.Width)
-            self.assertEqual(9134, document.DrawPages[0].getByIndex(1).Size.Height)
+            self.assertAlmostEqual(9134, document.DrawPages[0].getByIndex(1).Size.Height,delta=1)
 
             xEditWin = xImpressDoc.getChild("impress_win")
 
@@ -79,7 +79,7 @@ class ImpressDrawinglayerTest(UITestCase):
             self.assertEqual(25199, document.DrawPages[0].getByIndex(0).Size.Width)
             self.assertEqual(2629, document.DrawPages[0].getByIndex(0).Size.Height)
             self.assertEqual(12600, document.DrawPages[0].getByIndex(1).Size.Width)
-            self.assertEqual(4568, document.DrawPages[0].getByIndex(1).Size.Height)
+            self.assertAlmostEqual(4568, document.DrawPages[0].getByIndex(1).Size.Height,delta=1)
 
             self.assertIsNone(document.CurrentSelection)
 
@@ -110,7 +110,7 @@ class ImpressDrawinglayerTest(UITestCase):
             self.assertEqual(25199, document.DrawPages[0].getByIndex(0).Size.Width)
             self.assertEqual(2629, document.DrawPages[0].getByIndex(0).Size.Height)
             self.assertEqual(25199, document.DrawPages[0].getByIndex(1).Size.Width)
-            self.assertEqual(9134, document.DrawPages[0].getByIndex(1).Size.Height)
+            self.assertAlmostEqual(9134, document.DrawPages[0].getByIndex(1).Size.Height,delta=1)
             self.assertEqual(0, document.DrawPages[0].getByIndex(1).RotateAngle)
 
             xEditWin = xImpressDoc.getChild("impress_win")
@@ -121,7 +121,7 @@ class ImpressDrawinglayerTest(UITestCase):
             self.assertEqual(25199, document.DrawPages[0].getByIndex(0).Size.Width)
             self.assertEqual(2629, document.DrawPages[0].getByIndex(0).Size.Height)
             self.assertEqual(25199, document.DrawPages[0].getByIndex(1).Size.Width)
-            self.assertEqual(9134, document.DrawPages[0].getByIndex(1).Size.Height)
+            self.assertAlmostEqual(9134, document.DrawPages[0].getByIndex(1).Size.Height,delta=1)
             self.assertEqual(3000, document.DrawPages[0].getByIndex(1).RotateAngle)
 
             self.assertIsNone(document.CurrentSelection)
-- 
2.30.2

From f0972bd6b53c28cb2b927b362025d85f21cedcc4 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@rene-engelhard.de>
Date: Thu, 23 Dec 2021 16:42:21 +0100
Subject: [PATCH] work around rounding difference on i386 in
 descriptiveStatistics test

No idea why this only appears now, but:

======================================================================
FAIL: test_descriptive_statistics (descriptiveStatistics.descriptiveStatistics)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rene/tmp/autopkgtest-lxc.4j23khs6/downtmp/build.zIT/src/sc/qa/uitest/statistics/descriptiveStatistics.py", line 108, in te
st_descriptive_statistics
    self.assertEqual(round(get_cell_by_position(document, 0, 6, 10).getValue(),14) , 0.01524095329036)
AssertionError: 0.01524095329035 != 0.01524095329036

Change-Id: I496daf61a9542ea0da78e6092dd7070a2665ab0a
---
 sc/qa/uitest/statistics/descriptiveStatistics.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sc/qa/uitest/statistics/descriptiveStatistics.py b/sc/qa/uitest/statistics/descriptiveStatistics.py
index 13837804d287..b00bd9079a0d 100644
--- a/sc/qa/uitest/statistics/descriptiveStatistics.py
+++ b/sc/qa/uitest/statistics/descriptiveStatistics.py
@@ -107,7 +107,7 @@ class descriptiveStatistics(UITestCase):
             self.assertEqual(round(get_cell_by_position(document, 0, 6, 7).getValue(),11) , 139.49090909091)
             self.assertEqual(round(get_cell_by_position(document, 0, 6, 8).getValue(),12) , 11.810626955878)
             self.assertEqual(round(get_cell_by_position(document, 0, 6, 9).getValue(),13) , -1.4621677980825)
-            self.assertEqual(round(get_cell_by_position(document, 0, 6, 10).getValue(),14) , 0.01524095329036)
+            self.assertAlmostEqual(round(get_cell_by_position(document, 0, 6, 10).getValue(),14) , 0.01524095329036)
             self.assertEqual(get_cell_by_position(document, 0, 6, 11).getValue() , 31)
             self.assertEqual(get_cell_by_position(document, 0, 6, 12).getValue() , 26)
             self.assertEqual(get_cell_by_position(document, 0, 6, 13).getValue() , 57)
-- 
2.30.2

